ui/settings/src/commonMain/kotlin/social/androiddev/settings/SettingsContent.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of Dodo.
+ *
+ * Dodo is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * Dodo is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Dodo.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+package social.androiddev.settings
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import social.androiddev.common.composables.buttons.DodoButton
+import social.androiddev.common.composables.buttons.buttonColors
+
+@Composable
+fun SettingsContent(
+    component: SettingsComponent,
+    modifier: Modifier = Modifier,
+) {
+    SettingsContent(
+        modifier = modifier,
+        onLogout = component::logout,
+    )
+}
+
+@Composable
+internal fun SettingsContent(
+    onLogout: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Box(
+        modifier = modifier.fillMaxSize().padding(bottom = 64.dp),
```

Comment:
```suggestion
        modifier = modifier,
```

Reply:
`fillMaxSize` here would just override whatever size constraint was set by the caller. We should respect that size or use `then` modifier to only apply the size constraint if one wasn't specified. Same thing with padding

Reply:
- [ ] Fix ignoring outside/caller sizing constraints

Reply:
`fillMaxSize` does respect the incoming `maxWidth`/`maxHeight` constraint from the parent - it just fills up available space, instead of modifying it. Is that what you meant or am I missing something?

Reply:
Here's the scenario I am describing:

Given your `SettingsContent` internally takes in the passed in`modifier` and applies `.fillMaxSize().padding(bottom = 64.dp)`....

If the caller were to pass in `modifier.height(240.dp).width(120.dp)` or any other similar sizing modifier that _limits_ the size, your `.fillMaxSize` will effectively override that and force the composable to always take the full available space. 

You can quickly validate these via some preview functions

```kotlin

/**
* This should set the result size to fill maximum dimensions
*/

@Preview
@Composable
fun PreviewUseDefault() {
   SettingsContent(
      onLogout = {}.
   )
}

/**
* This should set the result size to h=240, w=120, but it will instead fill maximum dimensions
*/
@Preview
@Composable
fun PreviewThisShouldOverrideDefault() {
   SettingsContent(
      onLogout = {}.
      modifier = Modifier.height(240.dp).width(120.dp)
   )
}

```

Reply:
That makes sense! Indeed, `height`, `width` and `size` modifiers allow being overridden by incoming constraints (to be smaller or bigger), while `requiredHeight`, `requiredWidth`, and `requiredSize` set constraints to be exactly to the provided value.
Thanks for pointing this out